### PR TITLE
Feature/record 1120

### PR DESCRIPTION
**Pull request by juanlu-rti**
The example library generated was build without the CMAKE_BUILD_TYPE flag. This was causing the library to be release, but it was linked against the debug libraries in Connext. During runtime, the release libraries of Connext were used. This leads to unexpected behaviour. In that case it was leading to a crash (invalid pointer).

We should require that the user sets the CMAKE_BUILD_TYPE flag when using cmake to build the examples.

Internal issue: [RECORD-1120](https://jira.rti.com/browse/RECORD-1120).

### CODE REVIEW

File: examples/recording_service/pluggable_storage/c/README.md [hunk 1]
```diff
@@ -102,7 +102,7 @@ Then you can call CMake like this:
 
 ```bash
 cmake -DCONNEXTDDS_DIR=<connext dir> -DCMAKE_TOOLCHAIN_FILE=<toolchain file created above>
-        -DCONNEXTDDS_ARCH=<connext architecture> ..
+      -DCMAKE_BUILD_TYPE=Release -DCONNEXTDDS_ARCH=<connext architecture> ..
```
Comment (dmorales-rti) [inline examples/recording_service/pluggable_storage/c/README.md]: You should add -DBUILD_SHARED_LIBS=ON here. It is in the non-crosscompiled section.

File: examples/recording_service/pluggable_storage/cpp/README.md [hunk 2]
```diff
@@ -102,7 +102,7 @@ Then you can call CMake like this:
 
 ```bash
 cmake -DCONNEXTDDS_DIR=<connext dir> -DCMAKE_TOOLCHAIN_FILE=<toolchain file created above>
-        -DCONNEXTDDS_ARCH=<connext architecture> ..
+      -DCMAKE_BUILD_TYPE=Release -DCONNEXTDDS_ARCH=<connext architecture> ..
```
Comment (dmorales-rti) [inline examples/recording_service/pluggable_storage/cpp/README.md]: Same than the c/README, we are missing the -DBUILD_SHARED_LIBS=ON

File: examples/recording_service/pluggable_storage/c/README.md (hunk 1 above)
Comment (juanlu-rti) [reply examples/recording_service/pluggable_storage/c/README.md]: Will add.

File: examples/recording_service/pluggable_storage/cpp/README.md (hunk 2 above)
Comment (juanlu-rti) [reply examples/recording_service/pluggable_storage/cpp/README.md]: Added.

**Comment by iblancasa** [comment]
Please, could you modify the title and description of this pull request to show what are you trying to accomplish?

**Pull request merged**